kafka/protocol.py

```diff
@@ -52,7 +52,7 @@ def _encode_message_header(cls, client_id, correlation_id, request_key):
         """
         Encode the common request envelope
         """
-        return struct.pack('>hhih%ds' % len(client_id),
+        return struct.pack('>hhIh%ds' % len(client_id),
```

Comment:
the kafka protocol defines this field as a signed int32.  i think it's better to stick with that.